### PR TITLE
Collect controller

### PR DESCRIPTION
**Pull request by lukeoguro**
Initial collect controller at `/collect`. Just logging the necessary data right now.

### CODE REVIEW

**Comment by elguaposalsero** [comment]
testing webhook

**Comment by elguaposalsero** [comment]
testing

**Review by gcpinckert — APPROVED** [review]
Looks great!

File: controllers/collect.js [hunk 1]
```diff
@@ -1,2 +1,17 @@
 // Listening for requests from webhooks
 // Sends the requests to our PG and Mongo databases
+const express = require('express');
+const router = express.Router();
+const bodyParser = require('body-parser');
```
Comment (elguaposalsero) [inline controllers/collect.js:5]: Can we just require this in app.js so that we don't have to repeat it in every route

File: app.js [hunk 2]
```diff
@@ -1,15 +1,16 @@
 const express = require("express");
 const app = express();
-const bodyParser = require("body-parser");
```
Comment (elguaposalsero) [inline app.js:3]: Why did you remove this from app.js. Isn't it easier to just declare it once?

File: controllers/collect.js (hunk 1 above)
Comment (elguaposalsero) [inline controllers/collect.js:5]: Can we just require this in app.js so that we don't have to repeat it in every route

**Review by elguaposalsero — CHANGES_REQUESTED** [review]
Added some comments about bodyparser

File: controllers/collect.js (hunk 1 above)
Comment (lukeoguro) [reply controllers/collect.js:5]: I thought that too, but if you remember in full stack open, we end up actually creating mini-routers for each entity. More [here](https://expressjs.com/en/guide/routing.html#:~:text=Use%20the%20express.Router%20class%20to%20create%20modular%2C%20mountable%20route%20handlers.%20A%20Router%20instance%20is%20a%20complete%20middleware%20and%20routing%20system%3B%20for%20this%20reason%2C%20it%20is%20often%20referred%20to%20as%20a%20%E2%80%9Cmini%2Dapp%E2%80%9D.).

**Comment by elguaposalsero** [comment]
Test

File: controllers/collect.js (hunk 1 above)
Comment (lukeoguro) [reply controllers/collect.js:5]: For `bodyParser`, we need to parse the body as "raw" for the collect API, but json for the other routes, so we can't have a catch-all in `app.js`.

File: app.js (hunk 2 above)
Comment (lukeoguro) [reply app.js:3]: We need to have separate body-parsers for `collect` and the other APIs. `collect` needs to use "raw", while others use JSON, so you need to have two different `app.use(...)`

File: controllers/collect.js (hunk 1 above)
Comment (lukeoguro) [reply controllers/collect.js:5]: If you remember in full stack open, we end up extracting routes to mini-routers. More [here](https://expressjs.com/en/guide/routing.html#:~:text=Use%20the%20express.Router%20class%20to%20create%20modular%2C%20mountable%20route%20handlers.%20A%20Router%20instance%20is%20a%20complete%20middleware%20and%20routing%20system%3B%20for%20this%20reason%2C%20it%20is%20often%20referred%20to%20as%20a%20%E2%80%9Cmini%2Dapp%E2%80%9D.).

**Comment by elguaposalsero** [comment]
test

**Comment by elguaposalsero** [comment]
test

**Review by elguaposalsero — APPROVED** [review]
Looks good
